src/encoded/static/components/filegallery.js

```diff
@@ -2274,6 +2319,26 @@ class FileGalleryRendererComponent extends React.Component {
                 this.setState({ files: filteredFiles });
             }
             this.resetCurrentBrowser(null, allFiles);
+
+            // If new tab has been selected, we may need to update which assembly is chosen
+            if (updateAssemblyForTabChange) {
+                if (this.state.currentTab === 'tables') {
+                    // Always set the table assembly to be 'All assemblies'
+                    this.filterFiles('All assemblies', 'assembly');
+                } else if (this.state.currentTab === 'browser' || this.state.currentTab === 'graph') {
+                    // Determine available assemblies
+                    const assemblyList = this.setAssemblyList(allFiles);
+                    // Reset assembly filter if it is 'All assemblies' or is not in assemblyList
+                    // Assembly is required for browser / graph and available assemblies may be different for graph and browser
+                    // Do not reset if a particular assembly has already been chosen and it is an available option
+                    const currentAssembly = this.state.fileFilters.assembly[0];
+                    if (currentAssembly === 'All assemblies' || !(assemblyList[currentAssembly])) {
+                        // We want to get the assembly with the highest assembly number (but not 'All assemblies')
+                        const newAssembly = Object.keys(assemblyList).reduce((a, b) => (((assemblyList[a] > assemblyList[b]) && (a !== 'All assemblies')) ? a : b));
+                        this.filterFiles(newAssembly, 'assembly');
+                    }
```

Comment:
We talked about this part I guess. Might have to remember an actively selected assembly or something. Or possibly no one really cares that the chosen assembly doesn’t stick in some cases.

Reply:
I think it might be better to save a chosen assembly even if you click on the tables view in-between looking at the graph and the browser, but this is a little tricky and I'd rather talk about it more and make a new ticket.